File: apps/furniture-store-service-admin/src/api/product/ProductCreateInput.ts
import { CategoryWhereUniqueInput } from "../category/CategoryWhereUniqueInput";
import { OrderItemCreateNestedManyWithoutProductsInput } from "./OrderItemCreateNestedManyWithoutProductsInput";

export type ProductCreateInput = {
  category?: CategoryWhereUniqueInput | null;
  description?: string | null;
  name?: string | null;
  orderItems?: OrderItemCreateNestedManyWithoutProductsInput;
  price?: number | null;
  sku?: string | null;
  stock?: number | null;
};

File: apps/furniture-store-service-server/src/APIs/Product/Dtos/ProductCreateInput.cs
namespace FurnitureStoreService.APIs.Dtos;

public class ProductCreateInput
{
    public CategoryIdDto? Category { get; set; }

    public DateTime CreatedAt { get; set; }

    public string? Description { get; set; }

    public string? Id { get; set; }

    public string? Name { get; set; }

    public List<OrderItemIdDto>? OrderItems { get; set; }

    public double? Price { get; set; }

    public DateTime UpdatedAt { get; set; }
}
